frontend/src/container/OnboardingContainer/InfrastructureMonitoring/InfrastructureMonitoring.tsx

```diff
@@ -1,45 +1,154 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
 import './InfrastructureMonitoring.styles.scss';
 
+import cx from 'classnames';
 import { Code, Pre } from 'components/MarkdownRenderer/MarkdownRenderer';
+import { useEffect, useState } from 'react';
 import ReactMarkdown from 'react-markdown';
+import { trackEvent } from 'utils/segmentAnalytics';
 
-import InfraMonitoringDocs from './infraMonitoringDocs.md';
+import Header from '../common/Header/Header';
+import hostMetricsMonitoring from './md-docs/hostMetricsMonitoring.md';
+import k8sInfraMonitoringDocs from './md-docs/kubernetesInfraMonitoring.md';
+import otherMetrics from './md-docs/otherMetrics.md';
 
 export default function InfrastructureMonitoring({
 	activeStep,
 }: {
 	activeStep: number;
 }): JSX.Element {
-	const docsURL = 'https://signoz.io/docs/userguide/send-metrics-cloud/';
-	const heading = 'Send Metrics to SigNoz Cloud';
+	const [selectedInfraMetrics, setSelectedInfraMetrics] = useState('kubernetes');
+	const [selectedInfraMetricsDocs, setSelectedInfraMetricsDocs] = useState(
+		k8sInfraMonitoringDocs,
+	);
+
+	useEffect(() => {
+		// on metrics Type select
+		trackEvent('Onboarding: APM : Java', {
+			selectedInfraMetrics,
+		});
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [selectedInfraMetrics]);
+
+	const supportedInfraMetrics = [
```

Comment:
Can we shift this to constant so that it will not get created every time the component re-renders.

Reply:
We can take this in next PR.

Reply:
Sure.